src/endpoint/address.rs

```diff
@@ -133,7 +133,16 @@ impl FromStr for EndpointAddress {
         let host = url
             .host_str()
             .map(String::from)
+            .map(|value| {
+                // Strip ipv6 brackets
+                let len = value.len();
+                if len > 2 && value.starts_with('[') && value.ends_with(']') {
+                    return value[1..len - 1].to_string();
+                }
```

Comment:
Do the brackets not work when passed in? 🤔 Isn't that part of the address?

Reply:
If the brackets are still there in the host when it gets to

```
let host = host.parse::<AddressKind>().unwrap();
```

Then it parses it as as `AddressKind(Name("[::]"))` rather than `AddressKind(Ip(V6("::")))`, which is not what we want.

Reply:
Right, but I think a better solution would be to update the `FromStr` implementation of `AddressKind` to handle that instead of `EndpointAddress`, so that it's always correct, rather than `EndpointAddress::parse` and `AddressKind::parse` providing different results.

Reply:
Aaah gotcha. I will make that change 👍🏻